packages/cli/src/commands/export-dynamic-plugin/command.ts

```diff
@@ -74,5 +75,50 @@ export async function command(opts: OptionValues): Promise<void> {
     });
   }
 
+  await checkBackstageSupportedVersions(targetPath);
+
   await applyDevOptions(opts, rawPkg.name, roleInfo, targetPath);
 }
+
+async function checkBackstageSupportedVersions(targetPath: string) {
+  const targetPackageFile = path.join(targetPath, 'package.json');
+  const targetPackage = await fs.readJSON(targetPackageFile);
+  const supportedVersions: string | undefined =
+    targetPackage.backstage?.['supported-versions'];
+  const backstageJson = path.join(paths.targetRoot, '/backstage.json');
+  if (fs.existsSync(backstageJson)) {
+    const backstageVersion: string = (await fs.readJSON(backstageJson)).version;
+    if (supportedVersions) {
+      const singleVersionInSupportedVersions = semver.valid(
+        supportedVersions,
+        true,
+      );
+      const supportedVersionsRange = singleVersionInSupportedVersions
+        ? `~${supportedVersions}`
+        : supportedVersions;
+
+      if (!semver.subset(`~${backstageVersion}`, supportedVersionsRange)) {
+        const errorMessage = `The ${chalk.cyan('backstage.supported-versions')} field in the package descriptor is not compatible with the backstage version specified in the ${chalk.cyan('backstage.json')} file: ${chalk.cyan(supportedVersions)} vs ${chalk.cyan(backstageVersion)}.`;
```

Comment:
just a nit, but this string could probably be created on line 103 without the extra variable no?

Reply:
I also reuse it in line 107